docs/backup_and_restore.rst

```diff
@@ -123,67 +107,308 @@ Restoring
+
+   .. code:: sh
+
+       mkdir ~/Persistent/sd-restore-creds
+
+   Copy in the following files from ``~/Persistent/securedrop/install_files/ansible-base``:
+
+   -  ``tor_v3_keys.json``
+   - All ``.asc`` files (these correspond to your *Submission Public Key*, your OSSEC alerts public key, and (if configured) your journalist alerts public key)
+   - The file ``~/Persistent/securedrop/install_files/ansible-base/group-vars/all/site-specific``
+
+   Once you have preserved those files, rename the 
+   SecureDrop project root directory to ``securedrop.old``. 
+
+#. **Reinstall SecureDrop on new hardware:** Re-clone the SecureDrop repository.
+  
+   Copy the ``.asc`` files and the ``tor_v3_keys.json`` file into
+   ``~/Persistent/securedrop/install_files/ansible-base``
+   and the ``site-specific`` file to 
+   ``~/Persistent/securedrop/install_files/ansible-base/group-vars/all/site-specific``.
+
+   Prepare the new :doc:`servers <servers>`, and 
+   :doc:`reinstall SecureDrop <install>`. During the configuration 
+   stage (``./securedrop-admin sdconfig``), press 
+   "Enter" to use the values that are populated for you. Proceed through the 
+   installation, finishing with ``./securedrop-admin tailsconfig.``
+
+   If SSH-over-Tor is configured, run ``ssh app`` and ``ssh mon`` to add the
+   new onion URLs to your ``known_hosts`` file. 
+
+#. **Restore the backup**: Copy the backup archive (located in 
+   ``securedrop.old/install_files/ansible-base``) into the 
+   ``install_files/ansible-base`` directory and run
+
+   .. code:: sh
+            
+       ./securedrop-admin restore sd-backup-<your_backup_file>.tar.gz
+
+#. **Delete temporary files**: Remove any temporary files or directories
+   you created as you performed this migration, including the 
+   ``securedrop.old`` directory.
+
+.. _migrate_v2v3:
+
+Hardware Migration for Instances Using v2 + v3 Onion Services
+''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
+
+#. :ref:`Back up the existing installation <backing_up>`.
+
+#. **Preserve files and credentials**: Create a folder to
+   preserve important files from the old installation: 
+
+   .. code:: sh
+
+       mkdir ~/Persistent/sd-restore-creds
 
-* ``app-source-ths``
-* ``app-journalist-aths``
-* ``app-ssh-aths``
+   Copy in the following files from ``~/Persistent/securedrop/install_files/ansible-base``:
 
-Once SSH access to the servers has been established (or if using SSH over
-local network), Onion URLs for the *Source Interface* and *Journalist Interfaces*
-can be fetched using the installer:
+   -  ``tor_v3_keys.json``
+   - All ``.asc`` files (these correspond to your *Submission Public Key*, your OSSEC alerts public key, and (if configured) your journalist alerts public key
+   - The file ``~/Persistent/securedrop/install_files/ansible-base/group-vars/all/site-specific``
+   - (optional) ``app-ssh-aths`` and ``mon-ssh-aths``, if you wish to reinstate the ``app-ssh-legacy`` and ``mon-ssh-legacy`` commands
```

Comment:
mon isn't affected by the restore, so the monitor server after the restore will still have its  address from the fresh install. This means there's some divergence between how to handle v2 and v3 in this case. You can just retain the fresh install mon-ssh-aths in the v2 case, but you would need to update the original tor_v3_keys.json, adding in the mon-ssh keypair from the fresh install's tor_v3_keys.json, before running either tailsconfig (which will update the auth_private file) or install (which will update the client auth files on the server).